packages/amplify-appsync-simulator/src/data-loader/dynamo-db/index.ts

```diff
@@ -60,6 +63,37 @@ export class DynamoDBDataLoader implements AmplifyAppSyncSimulatorDataLoader {
     }
   }
 
+  private async deleteAllItems(): Promise<object | null> {
+    try {
+      const items = await this.getAllItems();
+      for await (const item of items) {
+        await this.client
+          .deleteItem({
+            TableName: this.tableName,
+            Key: { id: item.id },
+            ReturnValues: 'ALL_OLD',
+          }).promise();
+      }
```

Comment:
Code style suggestion. Instead of nesting awaits, I'd try to think about the promises you're trying to spin up, and whether they need to be sequenced for any particular reason. As written this code will delete one item at a time, but it may be more readable (and faster) to just do something like:
```ts
await Promise.all(items.map(item => this.client.deleteItem({
  TableName: this.tableName,
  Key: { id: item.id },
  ReturnValues: 'ALL_OLD',
})));
```

And kick off all deletes in parallel, then yield until all deletes to complete.

Note: for large tables, this could spin up a huge number of promises and requests to the client, so you instead may want to batch these by say 20 at a time (or some other tested value), or maybe we can actually chunk to a set of ids, and use [BatchWriteItems](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/DynamoDB.html#batchWriteItem-property) to delete in batches at the ddb level, instead of in client code.

Reply:
Tried to do it this way and records weren't getting deleted for some reason